app/models/spree/stock/estimator_decorator.rb

```diff
@@ -3,7 +3,14 @@ module Stock
     module EstimatorDecorator
       def shipping_rates(package, shipping_method_filter = ShippingMethod::DISPLAY_ON_FRONT_END)
 
-        if package.use_easypost?
+        # Only use easypost on the FrontEnd if the flag is set and the package
+        # flag allows for it to be used. Otherwise use the default spree methods.
+        # This allows for faster load times on the front end if we dont want to do dyanmic shipping
+
+        if package.use_easypost? &&
```

Comment:
Make this conditional statement into a private method with a meaningful name such as:
```
if use_dynamic_shipping(package)
...

private 

def use_dynamic_shipping(package)
  package.use_easypost? && package.use_easypost? && ShippingMethod::DISPLAY_ON_BOTH ||
  (Spree::Config[:use_easypost_on_frontend] && ShippingMethod::DISPLAY_ON_FRONT_END))
end
```

Reply:
I dont think this adds anything other than another than another function call. This is eventually going to get simplified so I think its okay to leave it as is for now. I was more looking for a review of the logic. Making sure nothing breaks like last time where this was reverted.

Reply:
do we care about `ShippingMethod::DISPLAY_ON_BACK_END`? it seems from your comment and the other conditionals that we only care about the front end here?

Reply:
This method is called by the backend so if backend is set then it must go through EP

Reply:
👍